pkg/distributor/distributor.go

```diff
@@ -1156,28 +1158,30 @@ type labelData struct {
 	hash uint64
 }
 
-func (d *Distributor) parseStreamLabels(vContext validationContext, key string, stream logproto.Stream) (labels.Labels, string, uint64, string, error) {
+func (d *Distributor) parseStreamLabels(vContext validationContext, key string, stream logproto.Stream) (labels.Labels, string, uint64, string, string, error) {
 	if val, ok := d.labelCache.Get(key); ok {
 		retentionHours := d.tenantsRetention.RetentionHoursFor(vContext.userID, val.ls)
-		return val.ls, val.ls.String(), val.hash, retentionHours, nil
+		return val.ls, val.ls.String(), val.hash, retentionHours, "", nil
```

Comment:
I think here we should resolve the policy as well
```suggestion
		return val.ls, val.ls.String(), val.hash, retentionHours, policy, nil
```